apps/gdalinfo_bin.cpp

```diff
@@ -148,8 +148,32 @@ MAIN_START(argc, argv)
 #ifdef __AFL_HAVE_MANUAL_CONTROL
             continue;
 #else
-        fprintf(stderr, "gdalinfo failed - unable to open '%s'.\n",
-                psOptionsForBinary->pszFilename);
+        bool bMaybeVector = false;
+        VSIStatBuf sStat;
+        if (VSIStat(psOptionsForBinary->pszFilename, &sStat) == 0)
```

Comment:
what about using GDALIdentifyDriver() instead and test if it is a vector driver? That would be more robust as not all drivers are extension specific, and that would work with non-file connection name

Same remark for ogrinfo

Reply:
Is there a potential cost to that? I worried about a bunch of filesystem or network activity just to generate a better error message, but that may be unfounded.

Reply:
That should be acceptable. Most network requests should already have bien cached. Abd this is an error code path so performance is less of a concern

Reply:
I've made this change